src/app_service/common/utils.nim

```diff
@@ -51,3 +51,11 @@ proc defaultDataDir(): string =
       else:
         targetDir
   absolutePath(joinPath(parentDir, "Status"))
+
+proc validateLink*(link: string): bool =
+  result = true
+  if link.len() != 0:
+    if not match(
+        link, re"[(http(s)?):\/\/(www\.)?a-zA-Z0-9@:%._\+~#=]{2,256}\.[a-z]{2,6}\b([-a-zA-Z0-9@:%_\+.~#?&//=]*)", 0):
```

Comment:
http URLs can also include UTF-8 text

Reply:
http://ko.wikipedia.org/wiki/%EC%9C%84%ED%82%A4%EB%B0%B1%EA%B3%BC:%EB%8C%80%EB%AC%B8

Reply:
My bad, they do show up as Unicode but must be percent encoded on input